README.md

```diff
@@ -247,6 +247,21 @@ TestClass.translate_hash(hash, tuple)
 => { 'first_name' => 'John', 'last_name' => 'Doe', 'maiden_name' => 'Doe' }
 ```
 
+#### NxtSupport::Crystalizer
+
+`NxtSupport::Crystalizer` crystallizes a shared value from an array of elements and screams in case the value is not  
+the same across the collection. This is useful in a scenario where you want to guarantee that certain objects share the same 
+attribute. Let's say you want to ensure that all users in your collection reference the same department, then the idea 
+is that you can crystallize the department from your collection. 
+
+```ruby
+NxtSupport::Crystalizer.new(collection: ['andy', 'andy']).call # => 'andy'
+NxtSupport::Crystalizer.new(collection: []).call # NxtSupport::Crystalizer::Error
+NxtSupport::Crystalizer.new(collection: ['andy', 'scotty']).call # NxtSupport::Crystalizer::Error  
+NxtSupport::Crystalizer.new(collection: insurances, with: :effective_at).call # => first insurance with shared effective_at  
```

Comment:
I don't want to be the party pooper :joy: but I'm not sure I like it returning the first element of the collection after crystallizing with `with`. :smile: Maybe I'm a bit too biased by the use cases (e.g. checking `Insurance::ChangeRequest`s for a common `effective_at`) but in these cases the return value would be useless because I'm getting back one more or less random element (not sure if one would ever do something with it). Either returning the whole collection or the crystallized value seems more suitable to me.
But if you like it I'm fine with it, since the crystallizer can be used quite flexibly.

Reply:
Gucci!